include/RAJA/exec-cuda/raja_cuda.hxx

```diff
@@ -144,14 +144,22 @@ struct cuda_exec_async : public cuda_exec<BLOCK_SIZE, true> {
 ///
 ///////////////////////////////////////////////////////////////////////
 ///
-template <size_t BLOCK_SIZE>
+template <size_t BLOCK_SIZE, bool Async = false>
```

Comment:
I realize this is old, but I made that change when I was doing the rest of the iterator rework.  It uses a bool parameter so that only a single host-side function is required to implement sync and async kernel launches, the single function just checks the bool template parameter to see if it should include a sync or not.